File: pbark/pbark/parser/keywords/logic_keyword_groups.py
from __future__ import annotations

from pbark.parser.keywords.keyword_registry import KeywordRegistry


class LogicKeywordGroups:
    def __init__(self, registry: KeywordRegistry) -> None:
        self.logical_and_words = registry.phrase_words("both", "and", "also")
        self.condition_and_words = registry.phrase_words("and")
        self.logical_or_words = registry.phrase_words("either")
        self.neither_words = registry.phrase_words("neither")
        self.nor_words = registry.phrase_words("nor")
        self.comparison_equal_words = registry.phrase_words("matches", "equals", "matching")
        self.comparison_not_equal_words = registry.phrase_words("different", "unlike", "differs")
        self.comparison_greater_words = registry.phrase_words("louder", "bigger", "more", "over", "above", "exceeds")
        self.comparison_less_words = registry.phrase_words("less", "fewer", "smaller", "under", "below", "quieter")
        self.comparison_at_least_words = registry.phrase_words("least")
        self.comparison_at_most_words = registry.phrase_words("most")
        self.not_words = registry.heard_words("not", "never")
        self.counts_words = registry.heard_words("counts")
        self.letters_words = registry.heard_words("letters", "length", "characters")
        self.inside_words = registry.heard_words("inside", "within", "through")
        self.plus_words = registry.heard_words("plus", "add")
        self.minus_words = registry.heard_words("minus", "spends")
        self.star_words = registry.phrase_words(
            "times", "multiply", "multiplies", "multiplied", "birthed", "births", "whelps", "whelped",
            "double", "doubles", "duplicates", "clones",
        )
        self.slash_words = registry.heard_words("broke", "divide", "divides", "halves")
        self.boolean_values = registry.heard_words("true", "false", "yes", "no", "yeah", "nope", "yep")
        self.null_values = registry.heard_words(
            "null", "nothing", "notreat", "emptybowl", "no-bone-here", "nobodyhome", "hungry", "starving",
            "empty", "barebowl", "emptykennel", "allgone", "vanished", "missing", "nowhere", "treatless",
            "boneless", "squeakless", "nofetch",
        )

File: src/main/java/dev/klomptech/jbark/parser/keywords/LogicKeywordGroups.java
package dev.klomptech.jbark.parser.keywords;

import java.util.List;

public final class LogicKeywordGroups {

  public final List<String> logicalAndWords;
  public final List<String> conditionAndWords;
  public final List<String> logicalOrWords;
  public final List<String> neitherWords;
  public final List<String> norWords;
  public final List<String> comparisonEqualWords;
  public final List<String> comparisonNotEqualWords;
  public final List<String> comparisonGreaterWords;
  public final List<String> comparisonLessWords;
  public final List<String> comparisonAtLeastWords;
  public final List<String> comparisonAtMostWords;
  public final List<String> notWords;
  public final List<String> countsWords;
  public final List<String> lettersWords;
  public final List<String> insideWords;
  public final List<String> plusWords;
  public final List<String> minusWords;
  public final List<String> starWords;
  public final List<String> slashWords;
  public final List<String> booleanValues;
  public final List<String> nullValues;

  public LogicKeywordGroups(final KeywordRegistry registry) {
    logicalAndWords = registry.phraseWords("both", "and", "also");
    conditionAndWords = registry.phraseWords("and");
    logicalOrWords = registry.phraseWords("either");
    neitherWords = registry.phraseWords("neither");
    norWords = registry.phraseWords("nor");
    comparisonEqualWords = registry.phraseWords("matches", "equals", "matching");
    comparisonNotEqualWords = registry.phraseWords("different", "unlike", "differs");
    comparisonGreaterWords =
        registry.phraseWords("louder", "bigger", "more", "over", "above", "exceeds");
    comparisonLessWords =
        registry.phraseWords("less", "fewer", "smaller", "under", "below", "quieter");
    comparisonAtLeastWords = registry.phraseWords("least");
    comparisonAtMostWords = registry.phraseWords("most");
    notWords = registry.heardWords("not", "never");
    countsWords = registry.heardWords("counts");
    lettersWords = registry.heardWords("letters", "length", "characters");
    insideWords = registry.heardWords("inside", "within", "through");
    plusWords = registry.heardWords("plus", "add");
    minusWords = registry.heardWords("minus", "spends");
    starWords =
        registry.phraseWords(
            "times",
            "multiply",
            "multiplies",
            "multiplied",
            "birthed",
            "births",
            "whelps",
            "whelped",
            "double",
            "doubles",
            "duplicates",
            "clones");
    slashWords = registry.heardWords("broke", "divide", "divides", "halves");
    booleanValues = registry.heardWords("true", "false", "yes", "no", "yeah", "nope", "yep");
    nullValues =
        registry.heardWords(
            "null",
            "nothing",
            "notreat",
            "emptybowl",
            "no-bone-here",
            "nobodyhome",
            "hungry",
            "starving",
            "empty",
            "barebowl",
            "emptykennel",
            "allgone",
            "vanished",
            "missing",
            "nowhere",
            "treatless",
            "boneless",
            "squeakless",
            "nofetch");
  }
}
